src/utils/permissions.ts

```diff
@@ -81,6 +86,45 @@ export async function botHasPermission(
   );
 }
 
+export async function botDependsPermission(
+  guildID: string,
+  permissions: Permission[],
+) {
+  const guild = await cacheHandlers.get("guilds", guildID);
+  if (!guild) throw new Error("GUILD_NOT_FOUND");
+
+  // Check if the bot is the owner of the guild, if it is, returns true
+  if (guild.ownerID === botID) return true;
+
+  const member = (await cacheHandlers.get("members", botID))?.guilds.get(
+    guildID,
+  );
+  if (!member) throw new Error("MEMBER_NOT_FOUND");
+
+  // The everyone role is not in member.roles
+  const permissionBits = [...member.roles, guild.id]
+    .map(function (id) {
+      return guild.roles.get(id)!;
+    })
+    // Remove any edge case undefined
+    .filter(function (r) {
```

Comment:
```suggestion
    .filter(r => {
```
Can you rename the argument to something else instead of just `r`?